src/hls-adapter.js

```diff
@@ -176,31 +200,48 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
    */
   load(startTime: ?number): Promise<Object> {
     if (!this._loadPromise) {
+      this._startTime = startTime;
       this._loadPromise = new Promise((resolve) => {
-        this._onLoadedMetadataCallback = this._onLoadedMetadata.bind(this, resolve);
-        this._videoElement.addEventListener(EventType.LOADED_METADATA, this._onLoadedMetadataCallback);
-        if (startTime) {
-          this._hls.startPosition = startTime;
-        }
-        if (this._sourceObj && this._sourceObj.url) {
-          this._hls.loadSource(this._sourceObj.url);
-          this._hls.attachMedia(this._videoElement);
-          this._trigger(EventType.ABR_MODE_CHANGED, {mode: this.isAdaptiveBitrateEnabled() ? 'auto' : 'manual'});
-        }
-      });
+        this._resolveLoad = resolve;
```

Comment:
deferred promise instead?

Reply:
Need to refactor our deferable promise cause the player and engine that calls `load` method expects to get regular promise and not our deferred one, so flow will complain.